python/GafferArnoldUI/ArnoldLightFilterUI.py

```diff
@@ -56,9 +56,17 @@ def __activateShader( parent ) :
 def __activateRoundness( parent ) :
 
 	if 'geometry_type' in parent :
-		return parent["geometry_type"].getValue() == "plane"
+		return parent["geometry_type"].getValue() == "plane" and not parent["shader"].getInput()
 	return True
 
+def __activateFalloff( parent ) :
+
+	if parent["shader"].getInput() :
```

Comment:
This isn't the behaviour I'm seeing out of Arnold (5.2). If I plug in a shader, it seems that the geometry type is ignored entirely, as well as the falloff. Do you see something different?

Reply:
Yeah, what I'm seeing is that when I use a "box", the `width_edge` and `height_edge` are used to scale the filter and the ramp is actually doing a rampy falloff. I'm not saying the edges make a lot of sense, but if they were set to something other than 0 and then I'd just go and deactivate the plugs, people were stuck with their previous settings that are now interpreted as scale...

Reply:
Maybe I haven't described what I'm seeing well enough. I'm not using a ramp, I'm plugging a noise into the shader input. As soon as I do that, all geometry/falloff controls appear to be ignored entirely.

```
import Gaffer
import GafferArnold
import GafferImage
import GafferOSL
import GafferScene
import IECore
import imath

Gaffer.Metadata.registerValue( parent, "serialiser:milestoneVersion", 0, persistent=False )
Gaffer.Metadata.registerValue( parent, "serialiser:majorVersion", 54, persistent=False )
Gaffer.Metadata.registerValue( parent, "serialiser:minorVersion", 0, persistent=False )
Gaffer.Metadata.registerValue( parent, "serialiser:patchVersion", 0, persistent=False )

__children = {}

__children["SceneReader"] = GafferScene.SceneReader( "SceneReader" )
parent.addChild( __children["SceneReader"] )
__children["SceneReader"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["distant_light"] = GafferArnold.ArnoldLight( "distant_light" )
parent.addChild( __children["distant_light"] )
__children["distant_light"].loadShader( "distant_light" )
__children["distant_light"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Group"] = GafferScene.Group( "Group" )
parent.addChild( __children["Group"] )
__children["Group"]["in"].addChild( GafferScene.ScenePlug( "in1", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Group"]["in"].addChild( GafferScene.ScenePlug( "in2", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Group"]["in"].addChild( GafferScene.ScenePlug( "in3", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Group"]["in"].addChild( GafferScene.ScenePlug( "in4", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Group"]["in"].addChild( GafferScene.ScenePlug( "in5", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Group"]["in"].addChild( GafferScene.ScenePlug( "in6", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Group"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Camera"] = GafferScene.Camera( "Camera" )
parent.addChild( __children["Camera"] )
__children["Camera"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Outputs"] = GafferScene.Outputs( "Outputs" )
parent.addChild( __children["Outputs"] )
__children["Outputs"]["outputs"].addChild( Gaffer.ValuePlug( "output1", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Outputs"]["outputs"]["output1"].addChild( Gaffer.StringPlug( "name", defaultValue = '', flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Outputs"]["outputs"]["output1"].addChild( Gaffer.BoolPlug( "active", defaultValue = True, flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Outputs"]["outputs"]["output1"].addChild( Gaffer.StringPlug( "fileName", defaultValue = '', flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Outputs"]["outputs"]["output1"].addChild( Gaffer.StringPlug( "type", defaultValue = '', flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Outputs"]["outputs"]["output1"].addChild( Gaffer.StringPlug( "data", defaultValue = '', flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Outputs"]["outputs"]["output1"].addChild( Gaffer.CompoundDataPlug( "parameters", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Outputs"]["outputs"]["output1"]["parameters"].addChild( Gaffer.CompoundDataPlug.MemberPlug( "quantize", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Outputs"]["outputs"]["output1"]["parameters"]["quantize"].addChild( Gaffer.StringPlug( "name", defaultValue = 'quantize', flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Outputs"]["outputs"]["output1"]["parameters"]["quantize"].addChild( Gaffer.IntVectorDataPlug( "value", defaultValue = IECore.IntVectorData( [ 0, 0, 0, 0 ] ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Outputs"]["outputs"]["output1"]["parameters"].addChild( Gaffer.CompoundDataPlug.MemberPlug( "displayHost", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Outputs"]["outputs"]["output1"]["parameters"]["displayHost"].addChild( Gaffer.StringPlug( "name", defaultValue = 'displayHost', flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Outputs"]["outputs"]["output1"]["parameters"]["displayHost"].addChild( Gaffer.StringPlug( "value", defaultValue = 'localhost', flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Outputs"]["outputs"]["output1"]["parameters"].addChild( Gaffer.CompoundDataPlug.MemberPlug( "remoteDisplayType", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Outputs"]["outputs"]["output1"]["parameters"]["remoteDisplayType"].addChild( Gaffer.StringPlug( "name", defaultValue = 'remoteDisplayType', flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Outputs"]["outputs"]["output1"]["parameters"]["remoteDisplayType"].addChild( Gaffer.StringPlug( "value", defaultValue = 'GafferImage::GafferDisplayDriver', flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Outputs"]["outputs"]["output1"]["parameters"].addChild( Gaffer.CompoundDataPlug.MemberPlug( "displayPort", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Outputs"]["outputs"]["output1"]["parameters"]["displayPort"].addChild( Gaffer.StringPlug( "name", defaultValue = 'displayPort', flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Outputs"]["outputs"]["output1"]["parameters"]["displayPort"].addChild( Gaffer.StringPlug( "value", defaultValue = '${image:catalogue:port}', flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Outputs"]["outputs"]["output1"]["parameters"].addChild( Gaffer.CompoundDataPlug.MemberPlug( "driverType", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Outputs"]["outputs"]["output1"]["parameters"]["driverType"].addChild( Gaffer.StringPlug( "name", defaultValue = 'driverType', flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Outputs"]["outputs"]["output1"]["parameters"]["driverType"].addChild( Gaffer.StringPlug( "value", defaultValue = 'ClientDisplayDriver', flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Outputs"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["InteractiveArnoldRender"] = GafferArnold.InteractiveArnoldRender( "InteractiveArnoldRender" )
parent.addChild( __children["InteractiveArnoldRender"] )
__children["InteractiveArnoldRender"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["StandardOptions"] = GafferScene.StandardOptions( "StandardOptions" )
parent.addChild( __children["StandardOptions"] )
__children["StandardOptions"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Catalogue"] = GafferImage.Catalogue( "Catalogue" )
parent.addChild( __children["Catalogue"] )
__children["Catalogue"]["images"].addChild( GafferImage.Catalogue.Image( "Image", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Catalogue"]["images"].addChild( GafferImage.Catalogue.Image( "Image1", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Catalogue"]["images"].addChild( GafferImage.Catalogue.Image( "Image2", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Catalogue"]["images"].addChild( GafferImage.Catalogue.Image( "Image3", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Catalogue"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["standard_surface"] = GafferArnold.ArnoldShader( "standard_surface" )
parent.addChild( __children["standard_surface"] )
__children["standard_surface"].loadShader( "standard_surface" )
__children["standard_surface"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["ShaderAssignment"] = GafferScene.ShaderAssignment( "ShaderAssignment" )
parent.addChild( __children["ShaderAssignment"] )
__children["ShaderAssignment"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["skydome_light"] = GafferArnold.ArnoldLight( "skydome_light" )
parent.addChild( __children["skydome_light"] )
__children["skydome_light"].loadShader( "skydome_light" )
__children["skydome_light"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["physical_sky"] = GafferArnold.ArnoldShader( "physical_sky" )
parent.addChild( __children["physical_sky"] )
__children["physical_sky"].loadShader( "physical_sky" )
__children["physical_sky"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["ArnoldOptions"] = GafferArnold.ArnoldOptions( "ArnoldOptions" )
parent.addChild( __children["ArnoldOptions"] )
__children["ArnoldOptions"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Plane"] = GafferScene.Plane( "Plane" )
parent.addChild( __children["Plane"] )
__children["Plane"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["light_blocker"] = GafferArnold.ArnoldLightFilter( "light_blocker" )
parent.addChild( __children["light_blocker"] )
__children["light_blocker"].loadShader( "light_blocker" )
__children["light_blocker"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["StandardAttributes"] = GafferScene.StandardAttributes( "StandardAttributes" )
parent.addChild( __children["StandardAttributes"] )
__children["StandardAttributes"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["PathFilter"] = GafferScene.PathFilter( "PathFilter" )
parent.addChild( __children["PathFilter"] )
__children["PathFilter"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["noise"] = GafferArnold.ArnoldShader( "noise" )
parent.addChild( __children["noise"] )
__children["noise"].loadShader( "noise" )
__children["noise"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["SceneReader"]["fileName"].setValue( '/home/john/dev/build/gaffer/resources/gafferBot/caches/gafferBot.scc' )
__children["SceneReader"]["__uiPosition"].setValue( imath.V2f( 18.4374352, -6.75042582 ) )
__children["distant_light"]["enabled"].setValue( False )
__children["distant_light"]["sets"].setValue( 'keyLight' )
__children["distant_light"]["__uiPosition"].setValue( imath.V2f( 5.43743658, -6.7503953 ) )
__children["Group"]["in"]["in0"].setInput( __children["distant_light"]["out"] )
__children["Group"]["in"]["in1"].setInput( __children["SceneReader"]["out"] )
__children["Group"]["in"]["in2"].setInput( __children["Camera"]["out"] )
__children["Group"]["in"]["in3"].setInput( __children["skydome_light"]["out"] )
__children["Group"]["in"]["in4"].setInput( __children["Plane"]["out"] )
__children["Group"]["in"]["in5"].setInput( __children["light_blocker"]["out"] )
__children["Group"]["__uiPosition"].setValue( imath.V2f( 19.9374313, -14.9144878 ) )
__children["Camera"]["transform"]["translate"].setValue( imath.V3f( 16.3724556, 30.5729942, 42.350853 ) )
__children["Camera"]["transform"]["rotate"].setValue( imath.V3f( -21.7724056, 22.3453407, 5.12264205e-06 ) )
__children["Camera"]["__uiPosition"].setValue( imath.V2f( 31.4374352, -6.74875402 ) )
__children["Outputs"]["in"].setInput( __children["StandardOptions"]["out"] )
__children["Outputs"]["outputs"]["output1"]["name"].setValue( 'Interactive/Beauty' )
__children["Outputs"]["outputs"]["output1"]["fileName"].setValue( 'beauty' )
__children["Outputs"]["outputs"]["output1"]["type"].setValue( 'ieDisplay' )
__children["Outputs"]["outputs"]["output1"]["data"].setValue( 'rgba' )
__children["Outputs"]["__uiPosition"].setValue( imath.V2f( 19.9374313, -67.6603317 ) )
__children["InteractiveArnoldRender"]["in"].setInput( __children["Outputs"]["out"] )
__children["InteractiveArnoldRender"]["__uiPosition"].setValue( imath.V2f( 19.9374313, -75.8244095 ) )
__children["StandardOptions"]["in"].setInput( __children["ArnoldOptions"]["out"] )
__children["StandardOptions"]["options"]["renderCamera"]["value"].setValue( '/group/camera' )
__children["StandardOptions"]["options"]["renderCamera"]["enabled"].setValue( True )
__children["StandardOptions"]["__uiPosition"].setValue( imath.V2f( 19.9374313, -59.4962654 ) )
Gaffer.Metadata.registerValue( __children["Catalogue"]["images"]["Image"], 'image:index', 0 )
__children["Catalogue"]["images"]["Image"]["fileName"].setValue( '/home/john/gaffer/projects/default/catalogues/blockers/536cb8d8f89f509c24b25fc1cc101dae.exr' )
Gaffer.Metadata.registerValue( __children["Catalogue"]["images"]["Image1"], 'image:index', 1 )
__children["Catalogue"]["images"]["Image1"]["fileName"].setValue( '/home/john/gaffer/projects/default/catalogues/blockers/c057bf8191faaa76448f58bf0fb563c4.exr' )
Gaffer.Metadata.registerValue( __children["Catalogue"]["images"]["Image2"], 'image:index', 2 )
__children["Catalogue"]["images"]["Image2"]["fileName"].setValue( '/home/john/gaffer/projects/default/catalogues/blockers/7c01548619d08090ce4d87054a39accf.exr' )
Gaffer.Metadata.registerValue( __children["Catalogue"]["images"]["Image3"], 'image:index', 3 )
__children["Catalogue"]["images"]["Image3"]["fileName"].setValue( '/home/john/gaffer/projects/default/catalogues/blockers/62d792a53a6195a9dc90aaa48a5206d2.exr' )
__children["Catalogue"]["imageIndex"].setValue( 4 )
__children["Catalogue"]["directory"].setValue( '${project:rootDirectory}/catalogues/${script:name}' )
__children["Catalogue"]["__mapping"].setValue( IECore.CompoundData({'Image':IECore.IntData( 0 ),'Image1':IECore.IntData( 1 ),'Image2':IECore.IntData( 2 ),'Image3':IECore.IntData( 3 ),'Image4':IECore.IntData( 4 )}) )
__children["Catalogue"]["__uiPosition"].setValue( imath.V2f( 41.0015259, -75.1076279 ) )
__children["standard_surface"]["__uiPosition"].setValue( imath.V2f( -8.97600555, -47.3448639 ) )
__children["ShaderAssignment"]["in"].setInput( __children["StandardAttributes"]["out"] )
__children["ShaderAssignment"]["shader"].setInput( __children["standard_surface"]["out"] )
__children["ShaderAssignment"]["__uiPosition"].setValue( imath.V2f( 19.9374313, -47.9472809 ) )
__children["skydome_light"]["transform"]["rotate"].setValue( imath.V3f( 0, -48.2799988, 0 ) )
__children["skydome_light"]["parameters"]["color"].setInput( __children["physical_sky"]["out"] )
__children["skydome_light"]["parameters"]["color"]["r"].setInput( __children["physical_sky"]["out"]["r"] )
__children["skydome_light"]["parameters"]["color"]["g"].setInput( __children["physical_sky"]["out"]["g"] )
__children["skydome_light"]["parameters"]["color"]["b"].setInput( __children["physical_sky"]["out"]["b"] )
__children["skydome_light"]["parameters"]["camera"].setValue( 0.0 )
__children["skydome_light"]["__uiPosition"].setValue( imath.V2f( 46.5734253, -6.75042534 ) )
__children["physical_sky"]["__uiPosition"].setValue( imath.V2f( 32.0785522, 2.96439648 ) )
__children["ArnoldOptions"]["in"].setInput( __children["ShaderAssignment"]["out"] )
__children["ArnoldOptions"]["options"]["aaSamples"]["enabled"].setValue( True )
__children["ArnoldOptions"]["__uiPosition"].setValue( imath.V2f( 19.9374313, -53.8322029 ) )
__children["Plane"]["transform"]["rotate"].setValue( imath.V3f( 270, 0, 0 ) )
__children["Plane"]["transform"]["scale"]["x"].setValue( 70167.46875 )
__children["Plane"]["transform"]["scale"]["y"].setInput( __children["Plane"]["transform"]["scale"]["x"] )
__children["Plane"]["transform"]["scale"]["z"].setInput( __children["Plane"]["transform"]["scale"]["x"] )
__children["Plane"]["__uiPosition"].setValue( imath.V2f( 61.2554169, -6.75042534 ) )
__children["light_blocker"]["transform"]["translate"].setValue( imath.V3f( -3.88295555, 5.37757587, 0 ) )
__children["light_blocker"]["transform"]["scale"].setValue( imath.V3f( 28.9176025, 9.44146633, 28.9176025 ) )
__children["light_blocker"]["parameters"]["geometry_type"].setValue( 'sphere' )
__children["light_blocker"]["parameters"]["density"].setValue( 1.0 )
__children["light_blocker"]["parameters"]["width_edge"].setValue( 1.100000023841858 )
__children["light_blocker"]["parameters"]["height_edge"].setValue( 0.20000000298023224 )
__children["light_blocker"]["parameters"]["axis"].setValue( 'y' )
__children["light_blocker"]["parameters"]["shader"].setInput( __children["noise"]["out"] )
__children["light_blocker"]["parameters"]["shader"]["r"].setInput( __children["noise"]["out"]["r"] )
__children["light_blocker"]["parameters"]["shader"]["g"].setInput( __children["noise"]["out"]["g"] )
__children["light_blocker"]["parameters"]["shader"]["b"].setInput( __children["noise"]["out"]["b"] )
__children["light_blocker"]["__uiPosition"].setValue( imath.V2f( 102.970245, -6.75042534 ) )
__children["StandardAttributes"]["in"].setInput( __children["Group"]["out"] )
__children["StandardAttributes"]["filter"].setInput( __children["PathFilter"]["out"] )
__children["StandardAttributes"]["attributes"]["linkedLights"]["enabled"].setValue( True )
__children["StandardAttributes"]["attributes"]["filteredLights"]["value"].setValue( 'defaultLights' )
__children["StandardAttributes"]["attributes"]["filteredLights"]["enabled"].setValue( True )
__children["StandardAttributes"]["__uiPosition"].setValue( imath.V2f( 19.9374313, -29.6995449 ) )
__children["PathFilter"]["paths"].setValue( IECore.StringVectorData( [ '/group/lightFilter' ] ) )
__children["PathFilter"]["__uiPosition"].setValue( imath.V2f( 34.3427048, -23.6166534 ) )
__children["noise"]["__uiPosition"].setValue( imath.V2f( 82.6487427, 2.23191619 ) )


del __children


```

Reply:
Ah, I haven't explained enough what I'm doing either :P I see where the differences we're seeing are coming from now. Try using an aiImage and see what happens then. But even when using a noise, the ramp _parameter_ still has an effect, no?

Reply:
Ah, yes, so aiImage doesn't behave except when using the box shape. I _think_ this is because it's relying on UVs being specified, and only the box mode seems to be specifying them. In other geometry modes, I can use the `aiImage.offset` parameter to shift around the UVs, entering `0.5, 0.5` to sample from the middle of the image, implying that those modes are simply setting the UVs to a constant `0,0`.

When using a noise, the `ramp` parameter is having an effect, yes. And the shape of that ramp is even dictated by the `geometry_type` parameter. But the shape itself isn't used to mask the result of the shader, and the `width_edge` and `width_height` parameters don't appear to be doing anything. Except when in box mode of course, where they adjust the UVs that the shader uses, presumably so that `0-1` fills the full size of the box, all the way to the soft edge.

I have no idea how to explain this to a user. Perhaps this?

- Disable `width_edge` and `width_height` when a shader is connected and the mode isn't box.
- Add a tooltip saying that UVs are only available to the shader in Box mode, so you'll need to shade based on P.